### PR TITLE
Fixed Data Widths for BRAM/FIFO

### PR DESCRIPTION
**Pull request by anaszameer**

### CODE REVIEW

**Review by jmannem-rs — APPROVED** [review]

**Review by jmannem-rs — APPROVED** [review]
approved.

**Comment by muhammadhamza15** [comment]
I am closing this PR due to the synchronization problem with the main branch.

File: specs/FIFO18KX2.yaml [hunk 1]
```diff
@@ -177,12 +177,12 @@ ports:
 
 parameters:
     DATA_WRITE_WIDTH1:
-      desc: FIFO data write width, FIFO 1 (1-18)
+      desc: FIFO data write width, FIFO 1 (9, 18)
       type: integer
       default: 18
       range: 1 .. 18
```
Comment (ashrafnisar) [inline specs/FIFO18KX2.yaml:183]: Why not define the range to be a comma seprated list of valid numbers. eg. range: [9, 18]?
is it do able?

File: specs/FIFO18KX2.yaml [hunk 2]
```diff
@@ -177,12 +177,12 @@ ports:
...
     DATA_READ_WIDTH1:
-      desc: FIFO data read width, FIFO 1 (1-18)
+      desc: FIFO data read width, FIFO 1 (9, 18)
       type: integer
       default: 18
       range: 1 .. 18
```
Comment (ashrafnisar) [inline specs/FIFO18KX2.yaml:188]: same comment as above

File: specs/TDP_RAM18KX2.yaml [hunk 3]
```diff
@@ -185,22 +185,22 @@ parameters:
       default: "{2048{1'b0}}"
       vector: 2048
     WRITE_WIDTH_A1:
-      desc: Write data width on port A, RAM 1 (1-18)
+      desc: Write data width on port A, RAM 1 (1, 2, 4, 9, 18)
       type: integer
       default: 18
       range: 1 .. 18
```
Comment (ashrafnisar) [inline specs/TDP_RAM18KX2.yaml:191]: againg. is it possible to define range: [1, 2, 4, 9, 18]

**Review by ashrafnisar — COMMENTED** [review]
I have added my comments. Please review and get back to me.

**Pull request closed without merging**